### PR TITLE
Modernize cardinality agg tests

### PR DESCRIPTION
**Pull request by not-napoleon**
This converts one test in the Cardinality tests to use (a new multi index version of) test case, which will eventually enable better memory accounting testing.  It also deletes a second test, which was only verifying that the aggregation context has the correct cache settings - that's not cardinality related, and testing it here is actually a PITA.  We're trying to push all aggregation context handling into one path, so we can make sure it's accessed correctly with regards to closing.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorTests.java [hunk 1]
```diff
@@ -618,47 +592,6 @@ public void testAsSubAggregation() throws IOException {
...
-        aggregator.postCollection();
-
-        final InternalCardinality cardinality = (InternalCardinality) aggregator.buildAggregation(0L);
-
-        assertEquals(10.0, cardinality.getValue(), 0);
-        assertEquals("cardinality", cardinality.getName());
-        assertTrue(AggregationInspectionHelper.hasValue(cardinality));
-
-        // Test that an aggregation not using a script does get cached
-        assertTrue(context.isCacheable());
```
Comment (not-napoleon) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorTests.java:655]: This assert is the only thing this test adds over the refactored test above.  I don't know where we should be testing that `SearchExecutionContext` is setting this correctly, but I don't think this is the right place. And having an aggregation context reference at this point is not good.  Let's discuss if this test adds enough value that we should re-create it somewhere closer to the code actually being tested.

Comment (nik9000) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorTests.java:655]: :+1:

I wonder if we should assert that all aggs are cacheable in test-case unless we have some kind of flag.

Comment (not-napoleon) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorTests.java:655]: That seems sensible.  I'll try that and see if anything breaks.

Comment (not-napoleon) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorTests.java:655]: So exactly two tests broke, both of which are sorting by a script so it makes sense they shouldn't be cached.  With #90149 merged, I am able to communicate that information to the right place in the test framework, with minimal intrusion on the rest of the code.  So I think we should be all set here.

**Comment by nik9000** [comment]
Nice!

**Review by nik9000 — APPROVED** [review]

**Comment by not-napoleon** [comment]
@elasticmachine update branch

**Pull request merged**